crates/fluvio-controlplane-metadata/src/smartmodule/spec.rs

```diff
@@ -174,3 +210,37 @@ mod base64 {
         base64::decode(base64.as_bytes()).map_err(serde::de::Error::custom)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::fs::read;
+
+    use super::SmartModuleWasm;
+
+    #[test]
+    fn validates_smartmodule_wasm_binary() {
+        let wasm_bytes =
+            read("tests/sm_ok.wasm").expect("Failed to find 'sm_ok.wasm' file for tests");
+        let result = SmartModuleWasm::validate_binary(wasm_bytes.as_slice());
+
+        assert!(result.is_ok());
+    }
+
+    #[test]
+    fn validate_input_to_be_wasm_file() {
+        let wasm_bytes =
+            read("tests/sm_k8_v1.yaml").expect("Failed to find 'sm_k8_v1.yaml' file for tests");
+        let result = SmartModuleWasm::validate_binary(wasm_bytes.as_slice());
+
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn validates_smartmodule_wasm_binary_corrupt() {
+        let wasm_bytes = read("tests/sm_need_more_data.wasm")
+            .expect("Failed to find 'sm_need_more_data.wasm' file for tests");
+        let result = SmartModuleWasm::validate_binary(wasm_bytes.as_slice());
+
+        assert!(result.is_err());
+    }
```

Comment:
For this test use case, I have removed many bytes from the end of the file to make sure the parsing is working as expected. Another test case is above for non WASM files.